src/main/java/org/broadinstitute/consent/http/models/Match.java

```diff
@@ -135,4 +140,12 @@ public void addFailureReason(String reason) {
             this.failureReasons.add(reason);
         }
     }
+
+    public static Match matchFailure(String consentId, String purposeId, List<String> failureReasons) {
+        return new Match(consentId, purposeId, false, false, true, MatchAlgorithm.V3, failureReasons);
+    }
+
+    public static Match matchSuccess(String consentId, String purposeId, DataUseMatchResultType match, List<String> failureReasons) {
+        return new Match(consentId, purposeId, Approve(match), Abstain(match), false, MatchAlgorithm.V3, failureReasons);
+    }
```

Comment:
In theory, failure reasons should always be an empty list in the success case.

Reply:
The failure reasons are extracted from the [response entity](https://github.com/DataBiosphere/consent/pull/1997/files#diff-b2fbd6a813de88461f1c1800da590a32cf7ac7312857be8a6b5566509240f799R174), so from what I can tell it is possible to have a MatchResult with [various associated reasons](https://github.com/DataBiosphere/consent-ontology/blob/develop/src/main/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherV3.java#L75), and these reasons will be sent as [failure reasons](https://github.com/DataBiosphere/consent-ontology/blob/develop/src/main/java/org/broadinstitute/dsde/consent/ontology/resources/MatchResource.java#L97) in the json.

So although the POST is successful and the `consent` match has not failed, it still may have one or more "failure reasons".